File: src/lib/types/create.ts
export type LicenseOption = {
  key: string;
  name: string;
};

export type CreateOptions = {
  owners: string[];
  gitignores: string[];
  licenses: LicenseOption[];
};

export type CreateRepoInput = {
  owner: string;
  name: string;
  description: string | null;
  private: boolean;
  addReadme: boolean;
  gitignore: string | null;
  license: string | null;
};

export type CreatedRepo = {
  fullName: string;
  htmlUrl: string;
  private: boolean;
};

File: src-tauri/src/models/create.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LicenseOption {
    pub key: String,
    pub name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateOptions {
    pub owners: Vec<String>,
    pub gitignores: Vec<String>,
    pub licenses: Vec<LicenseOption>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateRepoInput {
    pub owner: String,
    pub name: String,
    pub description: Option<String>,
    pub private: bool,
    pub add_readme: bool,
    pub gitignore: Option<String>,
    pub license: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreatedRepo {
    pub full_name: String,
    pub html_url: String,
    pub private: bool,
}
